template/pdk.gen.hpp.ejs

```diff
@@ -2,21 +2,24 @@
 #include <chrono>
 #include <cstddef>
 #include <expected>
+<% if (headerUsesJSON) { -%>
 #include <jsoncons/json.hpp>
+<% } -%>
 #include <memory>
 #include <span>
 #include <stdint.h>
 #include <string>
 #include <string_view>
+#include <unordered_map>
```

Comment:
This is not specific to the c++ bindgen, but this `#include` reminds me: it's possible to preserve input key order in some of our PDK languages today; e.g., using [`serde_json::Map`](https://docs.rs/serde_json/latest/serde_json/struct.Map.html) or `JSON.parse` in JS. There are some very specific situations where you'd want to preserve ordering– like outgoing request signatures.

I'm comfortable saying that the default parse in most non-JS languages produces unordered keys and directing folks to use JS or Rust pdks if they need ordering for now, though!

Reply:
Makes sense, we could also point them towards using an array of arrays of array of objects. If we need support for ordered maps, maybe there should be a way to specify in the schema? I'd prefer to not slow down most map usage by using an ordered map.

Reply:
Yeah – unordered should be a fine default! (I'm kind of just thinking out loud that we should document _a_ way to do it – `serde_json` exposes a feature flag that preserves key order at compile time, so we could expose that through the rust-bindgen's generated `Cargo.toml`. I don't think we'll have a ton of users that need it, though!)